jerry-ext/include/jerryscript-ext/args.h

```diff
@@ -0,0 +1,115 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRYX_ARGS_H
+#define JERRYX_ARGS_H
+
+#include <assert.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include "jerryscript.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif /* __cplusplus */
+
+typedef struct jerryx_arg_t jerryx_arg_t;
+typedef struct jerryx_arg_iterator_t jerryx_arg_iterator_t;
+
+/**
+ * Signature of the transform funtion.
+ */
+typedef jerry_value_t (*jerryx_arg_transform_func_t) (jerryx_arg_iterator_t *iterator_p);
+
+/**
+ * The structure of the jerry argument in arg transform.
+ * Each native argument should have one corresponding jerryx_arg_t item.
+ */
+struct jerryx_arg_t
+{
+  jerryx_arg_transform_func_t func; /**< the transform function for the native argument */
+  void *dest; /**< point to the native argument */
+  uintptr_t extra_info; /**< extra infomation for arg transform */
+};
+
+/**
+ * The argument iteration used for transform funtions.
+ */
+struct jerryx_arg_iterator_t
+{
+  const jerryx_arg_t *c_arg_p; /**< points to the current jerryx_arg_t */
+  const jerry_value_t *js_arg_p; /**< points to the current js argument */
+  const jerry_length_t  js_arg_cnt; /**< the total count of the js arguments */
+  jerry_length_t  js_arg_index; /**< the current index of the js argument */
+};
+
+jerry_value_t jerryx_arg_transform_all (const jerry_value_t this_val,
+                                        const jerry_value_t *js_arg_p,
+                                        const jerry_length_t js_arg_cnt,
+                                        const jerryx_arg_t *c_arg_p,
+                                        const jerry_length_t c_arg_cnt);
+
+jerry_value_t jerryx_arg_transform_args (const jerry_value_t *js_arg_p,
```

Comment:
what about sth more straight-forward? `jerryx_arg_transform_args` and `jerryx_arg_transform_args_with_this`